app/controllers/wpcc/your_details_controller.rb

```diff
@@ -48,6 +51,12 @@ def session_params
 
     def amend_session
       your_details_form_params.each { |key, value| session[key] = value }
+      session[:manually_opt_in] = manually_opt_in?
+    end
+
+    def manually_opt_in?
+      salary = your_details_form_params[:salary].to_i
```

Comment:
Is this not more responsibility of the form object? I would move there. And the check below? The controller does not sound the better place. Move to the form or create a one line class? Wdyt?